File: Python/B_ADHOC/Exercicio_1245.py
import re
while True:
  try:
      pares = 0
      myMap = {}
      n = int(input())
      a = ""
      for z in range(n):
        line = input().split(" ");
        numeroBota = int(line[0])
        
        if numeroBota in myMap:
          a = myMap.get(numeroBota)+line[1]
          myMap[numeroBota] = a
        else:
          myMap[numeroBota] = line[1]


      for x in myMap:
        D = len(re.findall("D", myMap[x]))
        E = len(re.findall("E", myMap[x]))
        if D>E and E!=0:
            pares+=E
        elif E>D and D!=0:
            pares+=D
        elif E==D:
            pares+=E
      print(pares)
  except EOFError:
    break

File: Javascript/A_NivelIniciante/Exercicio_1245.js
var input = require("./readFile.js").input;
var lines = input.split('\n');
while(true){
    var pares = 0
    var myMap = new Map();
    var n = lines.shift()*1
    var a = ""
    for(let z = 0;z<n;z++){
        var line = lines.shift().split(" ");
        var numeroBota = line[0]*1
        
        if(myMap.has(numeroBota)){
        a = myMap.get(numeroBota)+line[1].trim()
        myMap.set(numeroBota,a);
        continue;
        }
        myMap.set(numeroBota,line[1].trim());
        
    }

    myMap.forEach((value)=>
    {  
        var D = 0
        var E = 0
        if(value.match(/D/g)!=null){
            D = value.match(/D/g).length
        }
        if(value.match(/E/g)!=null){
            E = value.match(/E/g).length;
        }
        if(D>E && E!=0){
            pares+=E
        }else if(E>D && D!=0){
           pares+=D
        }else if(E==D){
            pares+=E
        }
    });
     console.log(pares)
     if(lines.length==0){
         break;
     }
}

